### ISSUE TITLE
Mangohud doesn't work with vkcube-wayland ?

### ISSUE THREAD

**Issue by SpidFightFR** [flightlessmango/MangoHud#1354]
**Describe the bug**

**List relevant hardware/software information**
- Void-Linux
- MangoHud 0.7.2
- RX 7900XTX

**To Reproduce**
1. execute `mangohud vkcube-wayland`
2. 
```
Cannot connect to wayland.
Exiting ...
```

**Expected behavior**
[should work, it works on my laptop with same specs]

**Screenshots**
N.A

**Additional context**
works with vkcube classic (xwayland).

**Comment by SpidFightFR** [comment]
for some reasons when i rollback to 0.7.0 release it works...

**Comment by flightlessmango** [comment]
can't repro this. works fine on my system with sway

**Comment by shmerl** [comment]
Works for me in Plasma 5.27.11

**Comment by SpidFightFR** [comment]
Hey there, i finally found the issue: amdvlk was taking a higher priority over mesa RADV.

So i had 2 solutions:

## Force RADV By default and force AMDVLK on a per-app basis (preferred method)
1. edit `/etc/environment`
2. add `VK_ICD_FILENAMES='/usr/share/vulkan/icd.d/radeon_icd.x86_64.json'`
3. reboot
4. If you need an app with AMDVLK, in my case, CP77 works better in path tracing with AMDVLK, then use `env VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/amd_icd64.json` on each game.

## Uninstall AMDVLK and only keep mesa radv.
1. [See with your distro Package manager]

**Comment by SpidFightFR** [comment]
The issue got back, despite amdvlk being absent.

Although this time, it happens only when i launch mangohud as an executable (e.g : `mangohud vkcube-wayland` doesn't work), however when i run it as an env var, it works (e.g: `env MANGOHUD=1 vkcube-wayland` works).

Something is wrong somewhere...

Also i do note it does works both ways on xwayland apps (e.g: regular `vkcube`).

And it doesn't work as well on other wayland native apps (e.g: Minecraft with Vulkanmod, which adds native wayland support to the game).

I hope this will help!

**Comment by shmerl** [comment]
If you want to check what's going on, run it with `VK_LOADER_DEBUG=all`.

**Comment by SpidFightFR** [comment]
> If you want to check what's going on, run it with `VK_LOADER_DEBUG=all`.

Hey, i had the occasion to try and i get this:

```
$ env VK_LOADER_DEBUG=all mangohud vkcube-wayland
INFO:             Vulkan Loader Version 1.3.268
Cannot connect to wayland.
Exiting ...
```

**Comment by shmerl** [comment]
Not sure, the only difference is that script is doing some LD_PRELOAD stuff which normal Vulkan loader isn't. You don't need it, so look into that script, it has some `DISABLE_LD_PRELOAD` and etc.

**Comment by SpidFightFR** [comment]
> Not sure, the only difference is that script is doing some LD_PRELOAD stuff which normal Vulkan loader isn't. You don't need it except for OpenGL, so look into that script, it has some `DISABLE_LD_PRELOAD` and etc.

yep can confirm, changing `disable_preload=false` to `disable_preload=true` makes it work.

**Comment by Etaash-mathamsetty** [comment]
is this fixed in mangohud-git ?

**Comment by SpidFightFR** [comment]
I'm sorry, i'm unable to test the git build, especially since this issue happens on production install... 😕

**Comment by shmerl** [comment]
It's a bug in a script, not in the library. Simply check if the script stopped doing the pointless LD preloading in Vulkan case.

**Comment by SpidFightFR** [comment]
> It's a bug in the script, not in the library. Simply check if the script stopped doing the pointless LD preloading in Vulkan case. Or don't use the script in the first place, it's unnecessary in Vulkan case anyway and is only useful for OpenGL.

when using the env var it doesn't seem to use the script, in that case: yes it does work.
Also as mentioned [here](https://github.com/flightlessmango/MangoHud/issues/1354#issuecomment-2293794335) i [modified the script](https://gist.github.com/SpidFightFR/eee1ab0a818e05ba00cd20e17420e4ed), adding the processes to the "disabled ld preloading list" also fixes the issue.

So in that case, the default for this LD preloading system should be disabled, and only enabled for OGL if it's only important for OGL...

**Comment by Etaash-mathamsetty** [comment]
> I'm sorry, i'm unable to test the git build, especially since this issue happens on production install... 😕

that's ok, by that I meant that it works on 0.7.2 for me

**Comment by flightlessmango** [comment]
Assuming this has been resolved
